snapcraft.yaml

```diff
@@ -31,51 +21,75 @@ apps:
     daemon: forking
     plugs:
       - network-bind
-  gnocchi-config-generator:
-    command: bin/gnocchi-config-generator
-  gnocchi-upgrade:
-    command: bin/gnocchi-upgrade
+  metricd:
+    command: snap-openstack gnocchi-metricd
+    daemon: simple
+    plugs:
+      - network
+  statsd:
+    command: snap-openstack gnocchi-statsd
+    daemon: simple
+    plugs:
+      - network
+  upgrade:
+    command: snap-openstack gnocchi-upgrade
+    plugs:
+      - network
+  change-sack-size:
+    command: snap-openstack gnocchi-change-sack-size
+    plugs:
+      - network
+  config-generator:
+    command: snap-openstack gnocchi-config-generator
+    plugs:
+      - network
 
 parts:
   gnocchi:
     plugin: python
     python-version: python2
-    source: http://tarballs.openstack.org/gnocchi/gnocchi-stable-3.1.tar.gz
+    source: https://pypi.debian.net/gnocchi/gnocchi-4.0.2.tar.gz
```

Comment:
We should probably make a ticket to get gnocchi 4.0.2 into the openstack tarball repo at some point in the near future. This works just fine, but will bypass some of the caching reverse proxy.

Reply:
I think that would be up to the upstream maintainer, and they've recently made a split away from openstack with gnocchi and no longer publish tarballs to openstack.org. The gnocchi project started with openstack but can run independently and can be useful without openstack, so they've moved many of their processes to be independent of openstack.

Reply:
Btw, I tried the github tarballs and they didn't work (they didn't include all the files iirc). And I think pypi tarballs don't necessarily persist long term. That's why I went with the debian pypi. It's what a lot of debian python packages use.